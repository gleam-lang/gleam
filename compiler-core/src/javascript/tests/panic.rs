use crate::{assert_js, assert_ts_def};

#[test]
fn bare() {
    assert_js!(
        r#"
fn go() {
  panic
}
"#,
    );
}

#[test]
fn panic_as() {
    assert_js!(
        r#"
fn go() {
  let x = "wibble"
  panic as x
}
"#,
    );
}

#[test]
fn bare_typescript() {
    assert_ts_def!(
        r#"
pub fn go() {
  panic
}
"#,
    );
}

#[test]
fn as_expression() {
    assert_js!(
        r#"
fn go(f) {
  let boop = panic
  f(panic)
}
"#,
    );
}

#[test]
fn pipe() {
    assert_js!(
        r#"
fn go(f) {
  f |> panic
}
"#,
    );
}

#[test]
fn sequence() {
    assert_js!(
        r#"
fn go(at_the_disco) {
  panic
  at_the_disco
}
"#,
    );
}

#[test]
fn case() {
    assert_js!(
        r#"
fn go(x) {
  case x {
    _ -> panic
  }
}
"#,
    );
}
