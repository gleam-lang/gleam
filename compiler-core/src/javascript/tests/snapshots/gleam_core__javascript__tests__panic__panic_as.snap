---
source: compiler-core/src/javascript/tests/panic.rs
expression: "\nfn go() {\n  let x = \"wibble\"\n  panic as x\n}\n"
---
----- SOURCE CODE

fn go() {
  let x = "wibble"
  panic as x
}


----- COMPILED JAVASCRIPT
import { makeError } from "../gleam.mjs";

function go() {
  let x = "wibble";
  throw makeError("panic", "my/mod", 4, "go", x, {})
}
